### ISSUE TITLE
Pulumi Golang stack always fails on Azure DevOps pipeline on hosted agents

### ISSUE THREAD

**Issue by giacomo-pati** [pulumi/pulumi-azure-native#764]
Pulumi Golang stack always fails on Azure DevOps pipeline on hosted agent (complete log attached):
From the log:
```log
...
+ go version
go version go1.16.3 linux/amd64
+ pulumi version
v2.25.2
+ pulumi pre --stack cicd --diff --refresh --non-interactive --color=never --logtostderr -v 9
Previewing update (cicd):
...
I0419 13:59:50.705386      29 langruntime_plugin.go:144] langhost[go].Run(pwd=/azapphost/pulumi/base,program=.,#args=0,proj=azapphost,stack=cicd,#config=3,dryrun=true) executing
go build github.com/pulumi/pulumi-azure-native/sdk/go/azure/network: /usr/local/go/pkg/tool/linux_amd64/compile: signal: killed
...
```
[azuredevops.log](https://github.com/pulumi/pulumi-azure-native/files/6336945/azuredevops.log)

Works fine locally on Win10 as well as within a similarly prepared Ubuntu 20.04 docker image (on Docker for Windows).
What worries me most is that it also doesn't run with exact same docker image as above on Azure DevOps pipeline :-/

From log above I suspect the go build command fails: go build github.com/pulumi/pulumi-azure-native/sdk/go/azure/network
## Expected behavior

Should run

## Current behavior

Fails as shown by the log above

## Steps to reproduce

1. Create a Pulumi project in Azure Git repo
2. Create a Azure DevOps pipeline that installs the tooling needed (Golang, Pulumi) and run command as shown above

**Comment by mikhailshilkov** [comment]
This probably sounds like out of memory. This provider is pretty heavy on memory at the moment, unfortunately. How much RAM do you have there?

**Comment by giacomo-pati** [comment]
> This probably sounds like out of memory. This provider is pretty heavy on memory at the moment, unfortunately. How much RAM do you have there?

Uh, haven't checked memory on those agents. Are you suggesting to go with the azure (classic) provider ATM?

**Comment by mikhailshilkov** [comment]
If that's possible, I suggest trying larger build agents to see if my guess is correct

**Comment by giacomo-pati** [comment]
According to
+ cat /proc/meminfo
MemTotal:        7121240 kB
MemFree:         4969092 kB
MemAvailable:    6290744 kB

**Comment by viveklak** [comment]
> > This probably sounds like out of memory. This provider is pretty heavy on memory at the moment, unfortunately. How much RAM do you have there?
> 
> Uh, haven't checked memory on those agents. Are you suggesting to go with the azure (classic) provider ATM?

You may also want to try upgrading to the just released azure-native 1.0 and pulumi CLI 3.0. We have made some improvements to the Go sdk which should help.

**Comment by giacomo-pati** [comment]
Yup, thx @viveklak ! Upgrading to Pulumi CLI 3.0.0, azure plugin 4.0.0, and azure-native plugin 1.0.0 helped ALOT!
Please close this issue!
